app/models/package_manager/pypi.rb

```diff
@@ -123,14 +123,14 @@ def self.known_versions(name)
         &.index_by { |v| v[:number] } || {}
     end
 
-    def self.dependencies(name, version, _mapped_project)
+    def self.dependencies(name, version, _mapped_project = nil)
       api_response = get("https://pypi.org/pypi/#{name}/#{version}/json")
       deps = api_response.dig("info", "requires_dist")
-      source_info = api_response.dig("releases", version)
+      source_info = api_response.fetch("urls", [])
       Rails.logger.warn("Pypi sdist (no deps): #{name}") unless source_info.any? { |rel| rel["packagetype"] == "bdist_wheel" }
 
       deps.map do |dep|
-        name, version = dep.split
+        name, version = dep.split(/ /, 2)
```

Comment:
What was the reason for this change?

Reply:
ohhh yes my bad - I meant to link the slack discussion on this: https://tidelift.slack.com/archives/C04GTTC0UJ0/p1674508566449779?thread_ts=1674504157.951459&cid=C04GTTC0UJ0

Reply:
TLDR: before we were saying definitively that a package had a certain number of dependencies, even if some of those dependencies only came with variants of the package. Now we include the information specifying whether they are only included with a variant.

Reply:
Allegedly (according to [this StackOverflow post](https://stackoverflow.com/questions/71165471/what-is-a-format-of-pypis-requires-dist-in-json-api)), these dependencies are in the [PEP 508](https://peps.python.org/pep-0508) format. I can't find an official PyPI source stating this though.

Looking at the tests in the [grammar section](https://peps.python.org/pep-0508/#complete-grammar), there are a couple of instances of valid PEP 508 dependencies where the part after a space would not specify a version, e.g.

```
name[quux, strange];python_version<'2.7' and platform_version=='2'
=> ["name[quux,", "strange];python_version<'2.7'"and platform_version=='2'"]
```

That's not the point of this change, but it's possible that the current (and this) implementation would create bad data

Reply:
That being said we should probably file something else to address it

Reply:
oof... 

According to [this](https://peps.python.org/pep-0508/#names), we should be able to use this regex to parse the name: `^([A-Z0-9]|[A-Z0-9][A-Z0-9._-]*[A-Z0-9])`. I suppose the rest of it we could provide in `requirements`?